README.md

```diff
@@ -29,6 +29,52 @@ To run the tutorials or the test suite locally, run the following to install the
 pip install -e .[dev]
 ```
 
+## Instanciation and Authentication
+
+There are several ways to provide a correct authentication using the SDK.
+
+```python
+from sdk import SDK
+
+group_id="your_group_id" # Replace this value by your group_id on the PASQAL platform.
+some_token="token"
+username="your_username" # Replace this value by your username or email on the PASQAL platform.
+password="your_password" # Replace this value by your password on the PASQAL platform.
+# Ideally, do not write this password in a script but provide in through the command-line or as a secret environment variable.
+
+""" Method 1: Username + Password
+    If you know your credentials, you can pass them to the SDK instance on creation.
+"""
+sdk = SDK(username=username, password=password, group_id=group_id)
+
+""" Method 2: Username only
+    If you only want to insert your username, but want a solution to have your password being secret
+    you can run the SDK without password. A prompt will then ask for your password
+"""
+sdk = SDK(username=username, group_id=group_id)
+> Please, enter your password:
+
+""" Method 3: Use a token.
+    If you already know your token, you can directly pass it as an argument.
+"""
+sdk = SDK(token_provider=token, group_id=group_id)
+```
+
+If you want to redefine the APIs used by the SDK, please, do the following.
+
+```python
+from sdk import SDK, Endpoints
+
+class MyNewEndpoints(Endpoints):
+    ...
+
+sdk = SDK(..., endpoints=MyNewEndpoints)
+```
+
+Doing so will result in you only overriding the endpoints you wish, and be able to use the exising ones.
```

Comment:
typo: existing

Overall I think this paragraph is not very clear, I'd suggest something along the lines of: "This enables you to target backend services running locally on your machine, or other environment like preprod or dev."